src/lib/command_line.ml

```diff
@@ -80,10 +81,25 @@ module Server = struct
     in
     Log.log t.log ~section:["server"; subsection] json
 
-  let save_job_list t =
-    Storage.Json.save_jsonable t.storage
-      ~path:["server"; "jobs.json"]
-      (`List (List.map t.jobs ~f:(fun j -> `String (Job.id j))))
+  let change_job_list t action =
+    let mutex =
+      match t.job_list_mutex with
+      | Some s -> s
+      | None ->
+        let m = Lwt_mutex.create () in
```

Comment:
Ah I got it; my confusion was in 1. forgetting that Lwt uses cooperative threading and 2. not thinking to check that `Storage.save_jsonable` used a `Lwt.bind`, which is where the problem could've occurred. Thank you for the clarification :)